compiler/rustc_trait_selection/src/traits/error_reporting/mod.rs

```diff
@@ -213,6 +220,95 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
         bug!();
     }
 
+    fn maybe_report_recursive_gat(
+        &self,
+        err: &mut DiagnosticBuilder<'_>,
+        parent: &PredicateObligation<'tcx>,
+        child: &PredicateObligation<'tcx>,
+    ) -> bool {
+        let (parent, child) =
+            match (parent.predicate.kind().skip_binder(), child.predicate.kind().skip_binder()) {
+                (ty::PredicateKind::Trait(head, ..), ty::PredicateKind::Trait(tail, ..)) => {
+                    (head, tail)
+                }
+                _ => return false,
+            };
+
+        if Some(parent.clone().def_id()) != self.tcx.lang_items().sized_trait() {
+            return false;
+        }
+
+        let parent_ty = parent.self_ty();
+        let child_ty = child.self_ty();
+        debug!(
+            "report_overflow_error_cycle: parent_ty = {:?}, child_ty = {:?}",
+            parent_ty, child_ty
+        );
+
+        let parent_kind = parent_ty.kind();
+        let child_kind = child_ty.kind();
+        debug!(
```

Comment:
FYI you can do `debug!(?parent_kind, ?child_kind)`. But, I feel like there are a lot of redundant debug statements here...(for example, we should get all the same info from the debug call above)